src/controllers/csi/metadata/sqlite.go

```diff
@@ -308,13 +309,18 @@ func (access *SqliteAccess) GetDynakube(dynakubeName string) (*Dynakube, error)
 	var tenantUUID string
 	var latestVersion string
 	var imageDigest string
-	var maxFailedMountAttempts int
+	var maxFailedMountAttempts *int
 
 	err := access.querySimpleStatement(getDynakubeStatement, dynakubeName, &tenantUUID, &latestVersion, &imageDigest, &maxFailedMountAttempts)
 	if err != nil {
 		err = errors.WithMessagef(err, "couldn't get dynakube, name '%s'", dynakubeName)
 	}
-	return NewDynakube(dynakubeName, tenantUUID, latestVersion, imageDigest, maxFailedMountAttempts), err
+
+	if maxFailedMountAttempts == nil {
+		maxFailedMountAttempts = address.Of(-1)
```

Comment:
shouldn't this be `defaultSqlMaxFailedMountAttempts`?
also if you have `ADD COLUMN MaxFailedMountAttempts INT DEFAULT ' + defaultSqlMaxFailedMountAttempts + ';` wouldn't it always return that default value when reading from the database ?

Reply:
also, setting the `DEFAULT` in the database feels scary a bit.

what happens if the default changes between releases or something, will the `ALTER` statement be able to handle that ?

Reply:
The ticket mentioned the column to be nullable.
The default value is only used if the value is not specified on an insert, if null is explicitly inserted the value will be null
If the value is null, it has been explicitly inserted, returning the default value in this case would make the column being nullable useless. I assume the intention is that if null, or -1, is returned, that there is effectively no maximum retry value and it should retry endlessly.
If the default is changed the alter command will handle that accordingly

Reply:
after talking about it, it turned out that you were completely correct, I will make the changes accordingly

Reply:
done